OpenMarket/OpenMarket/Utilities/NetworkAPI.swift

```diff
@@ -0,0 +1,37 @@
+//  Created by Aejong, Tottale on 2022/11/15.
+
+import Foundation
+
+enum NetworkAPI {
+    
+    static let scheme = "https"
+    static let host = "openmarket.yagom-academy.kr"
+    
+    case productList(query: [Query: String]?)
+    case product(productID: Int)
+    case healthCheck
+    
+    var urlComponents: URLComponents {
+        var urlComponents = URLComponents()
+        urlComponents.scheme = NetworkAPI.scheme
+        urlComponents.host = NetworkAPI.host
+        switch self {
+        case .productList(let query):
+            urlComponents.path = "/api/products"
+            if let query = query {
```

Comment:
swift 5.7에서는 아래와 같이 사용할 수 있어요.
```suggestion
            if let query {
```